ResearchKit/ActiveTasks/ORKWaitStepViewController.m

```diff
@@ -0,0 +1,218 @@
+                _activityIndicatorView.tintColor = [self tintColor];
+                [self addSubview:_activityIndicatorView];
+                [_activityIndicatorView startAnimating];
+                break;
+            case ORKProgressIndicatorMaskNone:
+            default:
+                break;
+        }
+        
+        [self setupConstraints];
+        [self setNeedsUpdateConstraints];
+    }
+    return self;
+}
+
+- (void)tintColorDidChange {
+    [super tintColorDidChange];
+    _progressView.progressTintColor = [self tintColor];
+    _activityIndicatorView.tintColor = [self tintColor];
+}
+
+- (void)setupConstraints {
+    
+    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-[_textLabel]-|"
+                                                                 options:NSLayoutFormatDirectionLeadingToTrailing
+                                                                 metrics:nil
+                                                                   views:NSDictionaryOfVariableBindings(_textLabel)]];
+    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[_textLabel]" options:NSLayoutFormatAlignAllCenterX metrics:nil views:NSDictionaryOfVariableBindings(_textLabel)]];
+    
+    if (_progressView) {
+        NSDictionary *screenMetric = @{@"progressWidth": [NSNumber numberWithFloat:([UIScreen mainScreen].bounds.size.width - 40.0)]};
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-(20.0)-[_progressView(progressWidth)]-(20.0)-|" options:NSLayoutFormatAlignAllBaseline metrics:screenMetric views:NSDictionaryOfVariableBindings(_progressView)]];
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:[_textLabel]-[_progressView]|" options:NSLayoutFormatAlignAllCenterX metrics:nil views:NSDictionaryOfVariableBindings(_progressView, _textLabel)]];
+    } else if (_activityIndicatorView) {
+        [self addConstraint:[NSLayoutConstraint constraintWithItem:_activityIndicatorView attribute:NSLayoutAttributeCenterX relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeCenterX multiplier:1.0 constant:0.0]];
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"H:|-(>=0)-[_activityIndicatorView]-(>=0)-|" options:NSLayoutFormatAlignAllBaseline metrics:nil views:NSDictionaryOfVariableBindings(_activityIndicatorView)]];
+        [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:[_textLabel]-[_activityIndicatorView]|" options:NSLayoutFormatAlignAllCenterX metrics:nil views:NSDictionaryOfVariableBindings(_activityIndicatorView, _textLabel)]];
+    }
+}
+
+#pragma mark Accessibility
+
+- (BOOL)isAccessibilityElement {
+    return YES;
+}
+
+- (NSString *)accessibilityLabel {
+    if (_progressView) {
+        return ORKAccessibilityStringForVariables(_textLabel.accessibilityLabel, _progressView.accessibilityLabel);
```

Comment:
Not read out for each interval, but only when the user taps the progress bar, it will read out whatever the progress is at that point in time.